.test-infra/dataproc/flink_cluster.sh

```diff
@@ -35,7 +35,7 @@
 #    HARNESS_IMAGES_TO_PULL='gcr.io/<IMAGE_REPOSITORY>/python:latest gcr.io/<IMAGE_REPOSITORY>/java:latest' \
 #    JOB_SERVER_IMAGE=gcr.io/<IMAGE_REPOSITORY>/job-server-flink:latest \
 #    ARTIFACTS_DIR=gs://<bucket-for-artifacts> \
-#    FLINK_DOWNLOAD_URL=https://archive.apache.org/dist/flink/flink-1.10.1/flink-1.10.1-bin-scala_2.11.tgz \
+#    FLINK_DOWNLOAD_URL=https://archive.apache.org/dist/flink/flink-1.12.3/flink-1.12.3-bin-scala_2.11.tgz \
```

Comment:
nit: I believe the currently released version of 1.12.x is `1.12.4`.
Unrelated: Do we have a policy about whic version to run on jenkins? Should it be the most recent one? Looks like we are changing from the last supported (1.10) to the second newest, is this intentional?

Reply:
Yes, I kept 1.12.3 because this is run in Dataproc and that's the currently supported version on Dataproc.

Reply:
Ah, understood.